testah-junit/src/main/java/org/testah/framework/report/jira/AbstractJiraRemoteLinkBuilder.java

```diff
@@ -0,0 +1,263 @@
+        icon.setUrl16x16(icon1Url16x16);
+        return icon;
+    }
+
+    /**
+     * Gets status.
+     *
+     * @param title    the title
+     * @param url      the url
+     * @param resolved the resolved
+     * @return the status
+     */
+    protected Status getStatus(final String title, final String url, final boolean resolved) {
+        final Status status = new Status();
+        status.setResolved(resolved);
+        status.setIcon(getIcon2(title, url));
+        return status;
+    }
+
+    /**
+     * Gets icon 2.
+     *
+     * @param title the title
+     * @param url   the url
+     * @return the icon 2
+     */
+    protected Icon2 getIcon2(final String title, final String url) {
+        Icon2 icon2 = new Icon2();
+        icon2.setLink(url);
+        icon2.setTitle(title);
+        icon2.setUrl16x16(icon2Url16x16);
+        return icon2;
+    }
+
+    /**
+     * Gets last test plan dto used, the testplan the testcase methods will use to get source and run info.
+     *
+     * @return the last test plan dto used
+     */
+    protected TestPlanDto getLastTestPlanDtoUsed() {
+        if (null == lastTestPlanDtoUsed) {
+            throw new RuntimeException("Unable to use getLastTestPlanDtoUsed() unless you "
+                    + " call either getRemoteLinkForTestPlanResult or getRemoteLinkForTestPlanResultKnownProblem or "
+                    + "set lastTestPlanDtoUsed with setLastTestPlanDtoUsed");
+        }
+        return lastTestPlanDtoUsed;
+    }
+
+    /**
+     * Sets last test plan dto used. Can be used to set the testplan the testcase methods will use to get source and run info.
+     *
+     * @param lastTestPlanDtoUsed the last test plan dto used
+     */
+    public void setLastTestPlanDtoUsed(final TestPlanDto lastTestPlanDtoUsed) {
+        this.lastTestPlanDtoUsed = lastTestPlanDtoUsed;
+    }
+
+    /**
+     * Gets run link to use. Can be used to set the testplan the testcase methods will use to get source and run info
+     *
+     * @return the run link to use
+     */
+    protected String getRunLinkToUse() {
+        return validateUrl(getLastTestPlanDtoUsed().getRunInfo().getRunLocation(), "Use-Envir-Param=param_runLocation");
+    }
+
+    /**
+     * Gets source link to use.
+     *
+     * @return the source link to use
+     */
+    protected String getSourceLinkToUse() {
+        String sourceUrl = validateUrl(TS.params().getSourceUrl(), "IssueGetting_param_sourceUrl");
+        return sourceUrl + (sourceUrl.endsWith("/") ? "" : "/") + getSourceWithSlash(getLastTestPlanDtoUsed().getSource());
+    }
+
+    /**
+     * Gets source with slash. For use with source code mgmt, turn the classpath seperator from dot to slash.
+     *
+     * @param source the source is the absolute classpath to the testplan class
+     * @return the source with slash
+     */
+    protected String getSourceWithSlash(final String source) {
+        if (StringUtils.isEmpty(source)) {
```

Comment:
You can simplify this by using https://commons.apache.org/proper/commons-lang/javadocs/api-3.6/src-html/org/apache/commons/lang3/StringUtils.html#line.5405